postgres/server/init.sls

```diff
@@ -20,6 +20,8 @@ postgresql-server:
     - pkgs: {{ pkgs | json }}
 {%- if postgres.use_upstream_repo == true %}
```

Comment:
This statement probably needs additional expression, the same as in `upstream.sls`:
```suggestion
{%- if 'pkg_repo' in postgres and postgres.use_upstream_repo == true %}
```
Because if "upstream" is flag is on, but there are no repo definition for particular distro, the requisite will fail.

Reply:
@sticky-note, I added the changes @vutny suggests and also added the same check in a few other places where `postgres.use_upstream_repo` was being checked but `pkg_repo` was not.

I think the changes in this PR should be OK for all the platforms, but I don't have a _FreeBSD_ around to try it.  Can you try it and give us your feedback?